core/consensus/common/src/evaluation.rs

```diff
@@ -19,11 +19,13 @@
 use super::MAX_BLOCK_SIZE;
 
 use parity_codec::Encode;
-use runtime_primitives::traits::{Block as BlockT, Header as HeaderT, As};
+use runtime_primitives::traits::{Block as BlockT, Header as HeaderT, One, CheckedConversion};
 use error_chain::{error_chain, error_chain_processing, impl_error_chain_processed,
 	impl_extract_backtrace, impl_error_chain_kind, bail};
 
-type BlockNumber = u64;
+// This is just a best effort to encode the number. None indicated that it's too big to encode
+// in a u128.
+type BlockNumber = Option<u128>;
```

Comment:
Maybe we could switch for stuff like this to some arbitrary length type.

Reply:
it's just for an error's display and therefore easier ways of achieving the same thing would be to either pre-render it as a `String` for the error or make it properly generic.